pkg/relay/ethereum/error_parse.go

```diff
@@ -0,0 +1,120 @@
+package ethereum
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+)
+
+type ErrorRepository map[[4]byte]abi.Error
+
+func CreateErrorRepository(abiPaths []string) (ErrorRepository, error) {
+	var errABIs []abi.Error
+
+	for _, dir := range abiPaths {
+		if err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() && strings.HasSuffix(info.Name(), ".json") {
+				data, err := os.ReadFile(path)
+				if err != nil {
+					return err
+				}
+
+				contractABI, err := abi.JSON(bytes.NewReader(data))
```

Comment:
Is this `err` handling required? @siburu

Reply:
Yes ... fixed now https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/39/commits/16a993702768fcfda7c5b9000820d2e55ae62026